cluster-autoscaler/cloudprovider/clusterapi/clusterapi_unstructured_test.go

```diff
@@ -0,0 +1,302 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterapi
+
+import (
+	"context"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/autoscaler/cluster-autoscaler/utils/units"
+)
+
+func TestSetSize(t *testing.T) {
+	initialReplicas := 1
+	updatedReplicas := int32(5)
```

Comment:
Why do we have to cast one but not the other?

Reply:
this came from the upstream change, i'll see if i can figure it out

Reply:
the main difference is that `updatedReplicas` is used more directly with the k8s api, whereas `initialReplicas` is used with internal test functions.

`initialReplicas` is _mostly_ used for passing the node count into the functions that create the test MachineSets and MachineDeployments. that said, there are 2 places where it is cast as an `int32` to compare with values from the api.

given the way the functions are written, we could clean this up in upstream and make them all `int32`s, but it is solidly in the code cleanliness refactor territory(and even that is debatable).